### PR TITLE
Filter fusion and structural varinat profiles while getting non mutation profiles

### PR DESCRIPTION
**Pull request by kalletlak**
Fix https://github.com/cBioPortal/cbioportal/issues/7024

### CODE REVIEW

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 1]
```diff
@@ -1211,11 +1212,7 @@ export class ResultsViewPageStore {
         ],
         invoke: () => {
             // we get mutations with mutations endpoint, all other alterations with this one, so filter out mutation genetic profile
-            const profilesWithoutMutationProfile = _.filter(
-                this.selectedMolecularProfiles.result,
-                (profile: MolecularProfile) =>
-                    profile.molecularAlterationType !== 'MUTATION_EXTENDED'
-            );
+            const profilesWithoutMutationProfile = getProfilesWithoutMutationProfile(this.selectedMolecularProfiles.result || []);
```
Comment (alisman) [inline src/pages/resultsView/ResultsViewPageStore.ts]: probably gauranteed that result will be array here.   i know ts complains so I think trailing ! is sufficient.

File: src/pages/resultsView/ResultsViewPageStoreUtils.ts [hunk 2]
```diff
@@ -610,3 +610,7 @@ export function fetchPatients(samples:Sample[]) {
         patientFilter
     });
 }
+
+export function getProfilesWithoutMutationProfile(molecularprofiles: MolecularProfile[]): MolecularProfile[] {
+    return molecularprofiles.filter(profile => !['MUTATION_EXTENDED', "FUSION", "STRUCTURAL_VARIANT"].includes(profile.molecularAlterationType));
```
Comment (alisman) [inline src/pages/resultsView/ResultsViewPageStoreUtils.ts]: can you just check to see if these values are available in an enum somewhere.  i'm pretty sure they are.

File: src/pages/resultsView/ResultsViewPageStoreUtils.ts [hunk 3]
```diff
@@ -610,3 +610,8 @@ export function fetchPatients(samples:Sample[]) {
         patientFilter
     });
 }
+
+export function getProfilesWithoutMutationProfile(molecularprofiles: MolecularProfile[]): MolecularProfile[] {
```
Comment (inodb) [inline src/pages/resultsView/ResultsViewPageStoreUtils.ts]: @kalletlak maybe more clear if the name is something like:

`excludeMutationAndSVProfiles`

`FUSION` is a subset of `STRUCTURAL_VARIANT` and will be deprecated in the future anyway

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 4]
```diff
@@ -1211,11 +1213,7 @@ export class ResultsViewPageStore {
         ],
         invoke: () => {
             // we get mutations with mutations endpoint, all other alterations with this one, so filter out mutation genetic profile
```
Comment (dippindots) [inline src/pages/resultsView/ResultsViewPageStore.ts]: Thanks @kalletlak for this quick fixing! Looks good to me!

Maybe can edit this comment to make it more clear like: 
"we get mutations with mutations endpoint, structural variants and fusions with structural variant endpoint, all other alterations with this one, so filter out mutation genetic profile and structural variant profiles "

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 5]
```diff
@@ -1270,12 +1268,7 @@ export class ResultsViewPageStore {
...
```
Comment (dippindots) [inline src/pages/resultsView/ResultsViewPageStore.ts]: Also here.

**Pull request merged**